app/src/main/java/com/gdg/pune/devfest19/ui/SpeakerDetailsActivity.kt

```diff
@@ -1,16 +1,126 @@
 package com.gdg.pune.devfest19.ui
 
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
+import android.view.View
+import android.widget.ImageButton
 import androidx.appcompat.app.AppCompatActivity
+import com.bumptech.glide.Glide
 import com.gdg.pune.devfest19.R
+import com.gdg.pune.devfest19.model.Speaker
+import kotlinx.android.synthetic.main.activity_speaker_details.*
+import kotlinx.android.synthetic.main.layout_social_links.*
 
 class SpeakerDetailsActivity : AppCompatActivity() {
 
+    var speaker = Speaker()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_speaker_details)
+        val extras = intent.extras
+        if (extras != null) {
+            speaker.name = extras.getString("speakerName")
```

Comment:
You are passing every data from intent. Just pass document id from intent and refer it here in `DocumentReference`. Refer code style of `SessionDetailsActivity`. It will be more easy.

Reply:
Wouldn't it result in re retrieving the query from firebase?

Reply:
Actually it's good practice in Firebase. If you are using `SnapshotListener` it loads recently loaded cached data.

Reply:
Okay great then, will do it!